Hardcodet.NotifyIcon.Wpf/Source/NotifyIconWpf/Interop/SystemInfo.cs

```diff
@@ -2,37 +2,41 @@
 
 namespace Hardcodet.Wpf.TaskbarNotification.Interop
 {
-  public static class SystemInfo
-  {
-    private static System.Windows.Point? dpiFactors;
-
-    private static System.Windows.Point? DpiFactors
+    public static class SystemInfo
     {
-      get
-      {
-        if (dpiFactors == null)        
-          using (var source = new HwndSource(new HwndSourceParameters()))
-            dpiFactors = new System.Windows.Point(source.CompositionTarget.TransformToDevice.M11, source.CompositionTarget.TransformToDevice.M22);        
-        return dpiFactors;
-      }
-    }
+        private static System.Windows.Point? dpiFactors;
 
-    public static double DpiXFactor
-    {
-      get
-      {
-        var factors = DpiFactors;
-        return factors.HasValue ? factors.Value.X : 1;
-      }
-    }
+        private static System.Windows.Point? DpiFactors
+        {
+            get
+            {
+                if (dpiFactors == null)
+                {
+                    using (var source = new HwndSource(new HwndSourceParameters()))
+                    {
+                        dpiFactors = new System.Windows.Point(source.CompositionTarget.TransformToDevice.M11, source.CompositionTarget.TransformToDevice.M22);
+                    }
+                }
+                return dpiFactors;
+            }
+        }
 
-    public static double DpiYFactor
-    {
-      get
-      {
-        var factors = DpiFactors;
-        return factors.HasValue ? factors.Value.Y : 1;
-      }
-    }         
-  }
-}
+        public static double DpiXFactor
+        {
+            get
+            {
+                var factors = DpiFactors;
+                return factors?.X ?? 1;
```

Comment:
Wouldn't sign off on this being more readable than the old-school evaluation, but fine by me 👍

Reply:
I actually considered changing more here, as the name of the class somehow doesn't represent the functionality it provides, and the code can be simplified.

What about this:

```
    /// <summary>
    /// This class is a helper to get the DPI factors
    /// </summary>
    public static class DpiFactor
    {
        private static readonly System.Windows.Point DpiFactors;

        static DpiFactor()
        {
            using (var source = new HwndSource(new HwndSourceParameters()))
            {
                if (source.CompositionTarget?.TransformToDevice != null)
                {
                    DpiFactors = new System.Windows.Point(source.CompositionTarget.TransformToDevice.M11, source.CompositionTarget.TransformToDevice.M22);
                    return;
                }
                DpiFactors = new System.Windows.Point(1, 1);
            }
        }

        /// <summary>
        /// Returns the DPI X Factor
        /// </summary>
        public static double FactorX => DpiFactors.X;

        /// <summary>
        /// Returns the DPI Y Factor
        /// </summary>
        public static double FactorY => DpiFactors.Y;
    }

```

Reply:
The calling code would be:

`return new Point { X = (int)(point.X / DpiFactor.FactorX), Y = (int)(point.Y / DpiFactor.FactorY) };`

Instead of:
`return new Point() { X = (int)(point.X / SystemInfo.DpiXFactor), Y = (int)(point.Y / SystemInfo.DpiYFactor) };`

Reply:
Let me know if you like that more, I will push another commit.

Reply:
The syntax is much cleaner. As far as the class name goes, it's debatable. From a purist view, your variant has the higher semantic cohesion.

OTOH, I would argue that the `SystemInfo` class name sort of documents that this is an attribute of the system, while the class in itself doesn't provide that context (it's the DPI factor of what?). Looking at this code the first time in years, this actually helped myself getting my head around it.

Bottom line: +1 for the cleaner properties, but I'd keep the class name. If you disagree, convince me :)

Reply:
I'm not here to convince you :grinning:, the name is okay.

Pushing a commit with keeping the `SystemInfo` name, which forces the property name to contain DPI, but changing the code.

Reply:
Perfect! Just drop me a note when you think you're done and I'll merge. Thanks for all the work you put in!